### PR TITLE
Improve Dependencies Makefiles

### PR DESCRIPTION
**Pull request by DavidMansolino**
This PR improves the dependencies Makefiles to make sure that if the dependency archive/directory is already existing, only to copy to ./lib (or other) is done and not the complete re-download + extraction.

This is particularly useful on Travis if we want to cache the `dependencies` folder.

### CODE REVIEW

File: dependencies/Makefile.mac [hunk 1]
```diff
@@ -69,56 +73,64 @@ $(WEBOTS_DEPENDENCY_PATH)/freetype-2.9/objs/.libs/libfreetype.6.dylib:
```
Comment (DavidMansolino) [inline dependencies/Makefile.mac]: @fabienrohrer, I have problems understanding how we use `libfreetype` on mac.
The archive is extracted in the `dependencies` folder and nothing is copied in any released folder.
If Webots is linked with `libfreetype.6.dylib` how does it works in the release? And if not, why is this dependency required?

Comment (DavidMansolino) [reply dependencies/Makefile.mac]: Ok I understand now, we are not linking with `libfreetype.6.dylib` as this Makefile may let think, but with `libfreetype.a`,  I will make it clearer in this Makefile.

**Comment by DavidMansolino** [comment]
The improvements of the dependencies makefiles allows now to cache the `dependencies` folder.

I took also advantage of this PR to improve the travis mechanism to use stages. This will make each part (e.g. tests and compilation for now) independent and speed up each independent part (by installing only the required apt/homebrew packages for examples).
Regular commit test: https://travis-ci.org/omichel/webots/builds/540676812?utm_source=github_status&utm_medium=notification
Daily test: https://travis-ci.org/omichel/webots/builds/540175722?utm_source=github_status&utm_medium=notification

**Comment by DavidMansolino** [comment]
test me

**Comment by DavidMansolino** [comment]
👍

**Comment by omichel** [comment]
:eyes:

**Pull request merged**